app.rb

```diff
@@ -0,0 +1,136 @@
+# include files
+require_relative './person'
+require_relative './book'
+require_relative './teacher'
+require_relative './student'
+require_relative './rental'
+require 'date'
+
+# app class
+class Apps
+  def initialize
+    @books = []
+    @rentals = []
+    @people = []
+  end
+
+  # list all books
+  def list_books
+    if @books.empty?
+      puts 'No book added yet.'
+      return
+    end
+    @books.map do |book|
+      puts "Title: \"#{book.title}\", Author: \"#{book.author}"
+    end
+  end
+
+  # list all people
+  def list_people
+    if @people.empty?
+      puts 'No person recorded yet.'
+      return
+    end
+    @people.map do |person|
+      puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  # create person
+  def create_person
+    print 'do you want to create a student (1) or a teacher (2)? [input the number]:'
+    option = gets.chomp
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'that is not valid input number'
+      return
+    end
+    puts 'Person created successfully'
+  end
```

Comment:
- [ ] In this `create_person` method, in case option 1 is entered (student), you need to ask if they have `parent_permission`. Please include that prompt as it is a vital part of the program, and I see you have skipped it.